src/lang.ts

```diff
@@ -10,95 +10,55 @@ function isObject(item: any): boolean {
 	return item && typeof item === 'object' && !Array.isArray(item) && !(item instanceof RegExp);
 }
 
-function copyArray(array: any[], kwArgs: CopyArgs): any[] {
+function copyArray(array: any[], inherited: boolean): any[] {
 	return array.map(function (item: any): any {
 		if (Array.isArray(item)) {
-			return copyArray(item, kwArgs);
+			return copyArray(item, inherited);
 		}
 
-		return isObject(item) ?
-			copy({
+		return !isObject(item) ?
+			item :
+			_mixin({
+				deep: true,
+				inherited: inherited,
 				sources: [ item ],
-				deep: kwArgs.deep,
-				descriptors: kwArgs.descriptors,
-				inherited: kwArgs.inherited,
-				assignPrototype: kwArgs.assignPrototype
-			}) :
-			item;
+				target: Object.create(null)
+			});
 	});
 }
 
-export function copy(kwArgs: CopyArgs): any {
-	const sources: Hash<any>[] = kwArgs.sources;
-	let target: any;
-
-	if (!sources.length) {
-		throw new RangeError('lang.copy requires at least one source object.');
-	}
-
-	if (kwArgs.assignPrototype) {
-		// create from the same prototype
-		target = Object.create(Object.getPrototypeOf(sources[0]));
-	}
-	else {
-		// use the target or create a new object
-		target = kwArgs.target || {};
-	}
+interface MixinArgs {
+	deep: boolean;
+	inherited: boolean;
+	sources: {}[];
+	target: {};
+}
 
-	for (let source of sources) {
-		if (kwArgs.descriptors) {
-			// if we are copying descriptors, use to get{Own}PropertyNames so we get every property
-			// (including non enumerables).
-			const names = (kwArgs.inherited ? getPropertyNames : Object.getOwnPropertyNames)(source);
+function _mixin(kwArgs: MixinArgs): {} {
+	const deep = kwArgs.deep;
+	const inherited = kwArgs.inherited;
+	const target = kwArgs.target;
 
-			for (let name of names) {
-				// get the descriptor
-				const descriptor = (kwArgs.inherited ?
-					getPropertyDescriptor : Object.getOwnPropertyDescriptor)(source, name);
-				let value = descriptor.value;
+	for (let source of kwArgs.sources) {
+		for (let key in source) {
+			if (inherited || hasOwnProperty.call(source, key)) {
```

Comment:
doing it this way is to guard against someone redefining `hasOwnProperty` on the object.  it ensures that this code uses the `hasOwnProperty` it expects to be using.